test/document.test.js

```diff
@@ -13927,6 +13927,32 @@ describe('document', function() {
     await savedDoc.save();
   });
 
+  it('avoids flattening objectids on insertMany (gh-14935)', async function() {
+    const TestSchema = new Schema(
+      {
+        professionalId: {
+          type: Schema.Types.ObjectId
+        },
+        firstName: {
+          type: String
+        },
+        nested: {
+          test: String
+        }
```

Comment:
```suggestion
        }
```
property seems unused and not relevant to this test